wagtail/contrib/modeladmin/options.py

```diff
@@ -547,7 +550,10 @@ def get_submenu_items(self):
         menu_items = []
         item_order = 1
         for modeladmin in self.modeladmin_instances:
-            menu_items.append(modeladmin.get_menu_item(order=item_order))
```

Comment:
Hi @Squarehost, rather than add conditionals here, I think a better approach might be to add support from the other side, by altering `__init__()` and `get_menu_item()` on `ModelAdminGroup` to support the passing of `parent` and `order`. We probably actually want the `order` passed to ModelAdminGroup.get_menu_item() to override the value set on it, so that the order specified in `items()` is respected.